src/library/autodj/autodjprocessor.cpp

```diff
@@ -505,6 +505,17 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
     while (true) {
+        // Incase we start off with < minimum tracks
+        bool randomQueueEnabled = (((m_pConfig->getValueString(
+                    ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+        int minAutoDJCrateTracks = m_pConfig->getValueString(
+                    ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+        int tracksToAdd = minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount();
+        while (randomQueueEnabled && (tracksToAdd > 0)) {
```

Comment:
I think it is a good idea to just transmit the requested track number by the signal and then the limit in 
AutoDJFeature::slotAddRandomTrack() can take care of it.